src/pykeen/models/nbase.py

```diff
@@ -249,6 +266,7 @@ def score_r(self, ht_batch: torch.LongTensor, slice_size: Optional[int] = None)
             t_indices=ht_batch[:, 1],
             slice_size=slice_size,
             slice_dim="r",
+            mode=mode,
         ).view(ht_batch.shape[0], self.num_relations)
```

Comment:
should `num_relations` ever be different based on mode?

Reply:
No - in our case relations are invariant at training/inference time. All relations in the inference graph are a subset of those seen at training, `|R_inf| <= |R_train|`. Conceptually, since it's a subset, one could argue that it should be restricted to the number of unique relations in the inference graph `|R_inf|`, but all existing inductive eval protocols in the literature assume it to be `|R_train|`